engine/src/witness/btc.rs

```diff
@@ -249,23 +279,37 @@ impl VoterApi<BitcoinEgressWitnessingES> for BitcoinEgressWitnessingVoter {
 		_settings: <BitcoinEgressWitnessingES as ElectoralSystemTypes>::ElectoralSettings,
 		properties: <BitcoinEgressWitnessingES as ElectoralSystemTypes>::ElectionProperties,
 	) -> Result<Option<VoteOf<BitcoinEgressWitnessingES>>, anyhow::Error> {
-		let BWElectionProperties { block_height: witness_range, properties: tx_hashes, .. } =
-			properties;
+		let BWElectionProperties {
+			block_height: witness_range,
+			properties: tx_hashes,
+			election_type,
+			..
+		} = properties;
 		let witness_range = BlockWitnessRange::try_new(witness_range).unwrap();
 
 		let mut txs = vec![];
-		// we only ever expect this to be one for bitcoin, but for completeness, we loop.
+		let mut response_block_hash: Option<bitcoin::BlockHash> = None;
 		tracing::info!("Witness range: {:?}", witness_range);
 		for block in BlockWitnessRange::<cf_chains::Bitcoin>::into_range_inclusive(witness_range) {
-			tracing::info!("Checking block {:?}", block);
-
-			let block_hash = self.client.block_hash(block).await?;
-
-			let block = self.client.block(block_hash).await?;
-
-			txs.extend(block.txdata);
+			match election_type {
+				BWElectionType::Optimistic => {
+					let block_hash = self.client.block_hash(block).await?;
+					let block = self.client.block(block_hash).await?;
+					response_block_hash = Some(block.header.hash);
+					txs.extend(block.txdata);
+				},
```

Comment:
It feels like we can improve the logical flow here a bit, we're currently matching on the `election_type` in every iteration of the loop, even though the election type cannot change between iterations. Also related: it would be nicer if we didn't have to have a mutable `respone_block_hash` variable... This currently only work correctly for the assumption that we only ever iterate once, I think.